### PR TITLE
Full Reindex for Alma from Dump File

### PR DESCRIPTION
**Pull request by tpendragon**
This uses the same strategy of downloading from itself as what currently
happens, but it has a test and happens outside of a rake task.

### CODE REVIEW

File: spec/services/alma/indexer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Alma::Indexer do
+  describe "#full_reindex" do
+    it "gets the latest full dump tar, unzips it, and indexes all the containing files" do
+      event = FactoryBot.create(:full_dump_event)
+      binding.pry
```
Comment (hackartisan) [inline spec/services/alma/indexer_spec.rb]: Is there more to this test somewhere?

**Review by hackartisan — CHANGES_REQUESTED** [review]

File: spec/services/alma/indexer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Alma::Indexer do
+  describe "#full_reindex!" do
+    it "gets the latest full dump tar files, unzips it, and indexes all the contained files" do
+      event = FactoryBot.create(:full_dump_event)
+      solr_url = ENV["SOLR_URL"] || "http://#{ENV['lando_marc_liberation_test_solr_conn_host']}:#{ENV['lando_marc_liberation_test_solr_conn_port']}/solr/marc-liberation-core-test"
```
Comment (hackartisan) [inline spec/services/alma/indexer_spec.rb:7]: To refactor into config later?

**Review by hackartisan — COMMENTED** [review]
Are we using both lando and solr wrapper?

Does this supercede #930 to close #865?

**Comment by tpendragon** [comment]
> Are we using both lando and solr wrapper?
> 
> Does this supercede #930 to close #865?

This doesn't set up Lando for Postgres.

It uses both Lando and Solr Wrapper, yes. Solr wrapper for CI because of the `.jar` files.

**Review by hackartisan — APPROVED** [review]

File: lib/tasks/dev.rake [hunk 3]
```diff
@@ -0,0 +1,48 @@
+namespace :server do
+  desc "Configures and starts dependent servers"
+  task initialize: :environment do
+    Rake::Task["db:create"].invoke
+    Rake::Task["db:migrate"].invoke
+  end
+
```
Comment (christinach) [inline lib/tasks/dev.rake:7]: This rake task doesn't work for me. If I drop the exisiting databases from postgres I'm not able to create a dev and a test db using the rake task.

File: spec/services/alma/indexer_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      stub_request(:get, "http://www.example.com/dump_files/spec%2Ffixtures%2Ffiles%2Falma%2Ffull_dump%2F1.xml.gz")
+        .to_return(status: 200, body: file_fixture("alma/full_dump/1.xml.gz").read, headers: {})
+      stub_request(:get, "http://www.example.com/dump_files/spec%2Ffixtures%2Ffiles%2Falma%2Ffull_dump%2F2.xml.gz")
+        .to_return(status: 200, body: file_fixture("alma/full_dump/2.xml.gz").read, headers: {})
+      solr = RSolr.connect(url: solr_url)
+      solr.delete_by_query("*:*")
+
```
Comment (christinach) [inline spec/services/alma/indexer_spec.rb:14]: line 13 fails in my local with `uninitialized constant Faraday::Error::ConnectionFailed`

**Review by christinach — COMMENTED** [review]
I added some comments. The rake task doesn't work in my local and the `rspec ./spec/services/alma/indexer_spec.rb` fails. I didn't block the pr in case it is an issue in my dev environment.

File: lib/tasks/dev.rake [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  desc "Stop lando solr and postgres servers."
+  task stop: :environment do
+    system("lando stop")
+  end
+
+  namespace :solr do
+    task start_solr_wrapper: :environment do
+      SolrWrapper.wrap(port: 8983, verbose: true, managed: true, download_dir: 'tmp', version: '8.4.1') do |solr|
```
Comment (christinach) [inline lib/tasks/dev.rake:22]: I can run this task locally only if I create the databases manually without using lando. The lando task to create the db in dev and test fails.

File: lib/tasks/dev.rake (hunk 3 above)
Comment (jrgriffiniii) [reply lib/tasks/dev.rake:7]: I am encountering these errors as well. I fixed this using https://github.com/pulibrary/marc_liberation/pull/930/files#diff-11dd86092c3c21bdb579844c51b327932d6364d851f3f3323af3e202d1758d8bR10

**Review by jrgriffiniii — CHANGES_REQUESTED** [review]
I encountered some errors with regards to the Lando support, and I needed to introduce a solution based upon https://github.com/pulibrary/marc_liberation/pull/930/files#diff-11dd86092c3c21bdb579844c51b327932d6364d851f3f3323af3e202d1758d8bR10 in order to resolve this.

**Comment by jrgriffiniii** [comment]
https://github.com/pulibrary/marc_liberation/compare/alma...alma-full-reindex-rebase-jrgriffiniii?expand=1 advances this to the point where everything is running for me locally.

**Comment by jrgriffiniii** [comment]
Following a successful review and merge of https://github.com/pulibrary/marc_liberation/pull/941, I shall close this.

**Comment by jrgriffiniii** [comment]
Deprecated by this solution being reintroduced with https://github.com/pulibrary/marc_liberation/pull/941

**Pull request closed without merging**